### PR TITLE
Refactor - Use interface for rekor and fulcio

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Refactored the implementation to accept the interface instead of the
  concrete implementation within the function.

- With this change `GenerateProvenance` can be tested with mocks. At the
  moment there aren't any unit tests for `GenerateProvenance`.

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

**Comment by naveensrinivasan** [comment]
@laurentsimon 👀

File: internal/builders/go/pkg/provenance.go [hunk 1]
```diff
@@ -18,13 +18,13 @@ import (
 	"context"
 	"encoding/hex"
 	"fmt"
+	"github.com/slsa-framework/slsa-github-generator/signing"
```
Comment (ianlewis) [inline internal/builders/go/pkg/provenance.go:21]: Move this import to the 3p block below. This block is stdlib packages.

**Review by ianlewis — APPROVED** [review]
Thanks!

**Review by laurentsimon — APPROVED** [review]
Thanks!

File: internal/builders/go/pkg/provenance.go (hunk 1 above)
Comment (naveensrinivasan) [reply internal/builders/go/pkg/provenance.go]: Thanks! Took care of it.

**Pull request merged**